Tasks/IISWebAppDeployment/DeployIISWebApp.ps1

```diff
@@ -1,4 +1,179 @@
-﻿param (
+﻿import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
+Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Deployment.Internal"
+Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Deployment.RemoteDeployment"
+
+
+Write-Verbose "Entering script DeployIISWebApp.ps1"
+
+function Get-HostName
+{
+    param(
+        [string]$protocol,
+        [string]$hostNameWithHttp,
+        [string]$hostNameWithSNI,
+        [string]$hostNameWithOutSNI,
+        [string]$sni
+    )
+    $hostName = [string]::Empty
+
+    if($protocol -eq "http")
+    {
+        $hostName = $hostNameWithHttp
```

Comment:
actually if protocol is not http and that means it is https. In this case if sni is enabled hostname is mandatory to be able to achieve that we have add two parameters in json, so appropriately reading the hostname input from UI